src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -281,15 +329,20 @@ private void EnsureGlobbingUrlBuilder()
                    .Append("</script>");
         }
 
-        private void AppendSrc(TagHelperContent content, string srcKey, string srcValue)
+        private bool ShouldAddFileVersion()
         {
-            // Append src attribute in the original place and replace the content the fallback content
-            // No need to encode the key because we know it is "src".
-            content.Append(" ")
-                   .Append(srcKey)
-                   .Append("=\\\"")
-                   .Append(HtmlEncoder.HtmlEncode(srcValue))
-                   .Append("\\\"");
+            return FileVersion ?? false;
```

Comment:
I tried that. It gets more confusing when you try to use this along with ternary operators. Hence I separated this. I think this is a lot more readable.